app_legacy/Helpers/database/leaderboard.php

```diff
@@ -22,34 +26,19 @@ function SubmitLeaderboardEntry($user, $lbID, $newEntry, $validation): array
     if ($dbResult !== false) {
         $lbData = mysqli_fetch_assoc($dbResult);
 
-        $lbID = $lbData['LeaderboardID'];
-        $gameID = $lbData['GameID'];
-        $lbTitle = $lbData['Title'];
-        $lowerIsBetter = $lbData['LowerIsBetter'];
-        settype($lowerIsBetter, 'integer');
+        $lowerIsBetter = (int) $lbData['LowerIsBetter'];
 
         $scoreFormatted = GetFormattedLeaderboardEntry($lbData['Format'], $newEntry);
 
         $retVal['LBData'] = $lbData;
-        settype($retVal['LBData']['LeaderboardID'], 'integer');
-        settype($retVal['LBData']['GameID'], 'integer');
-        settype($retVal['LBData']['LowerIsBetter'], 'integer');
+        $retVal['LBData']['LeaderboardID'] = (int) $retVal['LBData']['LeaderboardID'];
+        $retVal['LBData']['GameID'] = (int) $retVal['LBData']['GameID'];
+        $retVal['LBData']['LowerIsBetter'] = (int) $retVal['LBData']['LowerIsBetter'];
 
         $retVal['Score'] = $newEntry;
-        settype($retVal['Score'], 'integer');
```

Comment:
`$retVal['Score']` needs to be an integer. The simple parser in the rcheevos code gets confused by the quotation mark.

Reply:
I'm not sure how this can be a non-integer at this point. When trying to cast it to an int explicitly my IDE complains that it's already an integer:

`$retVal['Score'] = (int) $newEntry;`

I'm missing something.

Reply:
`sanitize_sql_inputs` (line 16) is changing it.

Reply:
oh 🤦 thank you! forgot to remove the integers from the sql sanitization in this case.
Fixed in https://github.com/RetroAchievements/RAWeb/pull/1393/commits/08d2381ca160369cce6cce76c808ed8151abcdf0